testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java

```diff
@@ -0,0 +1,130 @@
+
+@RunAsClient
+@RunWith(Arquillian.class)
+@ServerSetup(StartWorkersInSuspendedModeTestCase.ServerSetupTask.class)
+public class StartWorkersInSuspendedModeTestCase extends AbstractClusteringTestCase {
+
+    private static final String MODULE_NAME = StartWorkersInSuspendedModeTestCase.class.getSimpleName();
+    private static final String DEPLOYMENT_NAME = MODULE_NAME + ".war";
+    public static final long STATUS_REFRESH_TIMEOUT = 30_000;
+    public static final int LB_OFFSET = 500;
+
+    @Deployment(name = DEPLOYMENT_1, testable = false, managed = false)
+    @TargetsContainer(NODE_1)
+    public static WebArchive deployment() {
+        WebArchive pojoWar = ShrinkWrap.create(WebArchive.class, DEPLOYMENT_NAME);
+        pojoWar.add(new StringAsset("Hello World"), "index.html");
+        return pojoWar;
+    }
+
+    public StartWorkersInSuspendedModeTestCase() {
+        super(new String[] { NODE_1, LOAD_BALANCER_1 }, new String[]{DEPLOYMENT_1});
+    }
+
+    @Test
+    public void testWorkerNodeIsRegisteredStopped() throws Exception {
+        ServerReload.executeReloadAndWaitForCompletion(TestSuiteEnvironment.getModelControllerClient(), ServerReload.TIMEOUT,
+                                                       false, true,
+                                                       null,
+                                                       -1, null);
+
+        assertWorkerNodeContextIsStopped();
+    }
+
+    private void assertWorkerNodeContextIsStopped() throws Exception {
+
+        ModelNode op = createOpNode("subsystem=undertow/configuration=filter/mod-cluster=load-balancer/balancer=mycluster/node=" + NODE_1,
+                                    READ_RESOURCE_OPERATION);
+        op.get(ADDRESS).add("context", "/" + MODULE_NAME);
+        op.get(RECURSIVE).set(true);
+        op.get(INCLUDE_RUNTIME).set(true);
+
+        final ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient(
+           null,
+           TestSuiteEnvironment.getServerAddress(),
+           TestSuiteEnvironment.getServerPort() + LB_OFFSET);
+
+        // might need to wait for mod_cluster nodes to be registered
+        long start = System.currentTimeMillis();
+        ModelNode modelNode = null;
+        while (System.currentTimeMillis() - start < STATUS_REFRESH_TIMEOUT) {
+            modelNode = client.execute(op);
+            if (modelNode.has(RESULT) && modelNode.get(RESULT).has(STATUS)) {
+                break;
+            }
+            Thread.sleep(100);
+        }
+
+        Assert.assertEquals(SUCCESS, modelNode.get(OUTCOME).asString());
+        Assert.assertEquals("stopped", modelNode.get(RESULT).get(STATUS).asString());
+    }
+
+    static class ServerSetupTask extends CLIServerSetupTask {
+        public ServerSetupTask() {
+
+            /*
+            Set up balancer using deprecated proxy-list option in modcluster subsystem
+             */
```

Comment:
Just a nitpick, this comment is just false. Possibly a copy paste issue.

Reply:
Thanks, removed the comment now. I missed it when changing the setup method earlier on.